astropy/utils/data.py

```diff
@@ -952,12 +952,16 @@ def check_free_space_in_dir(path, size):
 
 
 def _download_file_from_source(source_url, show_progress=True, timeout=None,
-                               remote_url=None, cache=False, pkgname='astropy'):
+                               remote_url=None, cache=False, pkgname='astropy',
+                               http_headers={'User-Agent': 'astropy',
```

Comment:
I wasn't expecting this custom header to be the default all the way down here.

Reply:
For this internal function called only from one place there isn't actually a need for defaults at all, in fact it probably shouldn't have them (so as to catch calls that omit an argument); they could also be keyword-only. That's my fault. But passing a mutable default argument is bad form.

Reply:
Good point. I moved this construct to `download_files`. Still the relevant question of whether it's *desirable*, though.

Reply:
I propose set `http_headers` to `None` in `download_file`. It should only be set in `iers.py` because it is necessary there. The rest should be done in a case-by-case basis from higher level calls.